### DISCUSSION TITLE
what-is-ebpf/ebpf-programs/

### DISCUSSION THREAD

**Discussion by giscus** [rootsongjc/cloud-native-library#104] [Announcements]
# what-is-ebpf/ebpf-programs/

在这一章中，让我们来谈谈编写 eBPF 代码。我们需要考虑在内核中运行的 eBPF 程序本身，以及与之交互的用户空间代码。 内核和用户空间代码 首先，你可以用什么编程语言来编写 eBPF 程序？ 内核接受字节码形式的 eBPF 程序 1。人工编写这

https://lib.jimmysong.io/what-is-ebpf/ebpf-programs/

**Comment by zonghaishang** [comment]
`在 eBPF 程序的例子中，我选择了 opennoop` 中存在拼写错误，`opennoop`应该为`opensnoop`

**Reply by rootsongjc** [reply]
在下文中还有一些 opensnoop 中漏掉了字母 s，我一并修改了。
